### PR TITLE
When updating reports, delete those that no longer apply

### LINKED ISSUE #241: Enable removal of outdated reports
As noted in a [discussion on a false positive report](https://github.com/w3c/strudy/issues/7#issuecomment-1324650052), the script that updates existing reports need to cater for situations where an existing report needs not to be updated but removed because none of its listed anomalies remain open.

### PR DESCRIPTION
**Pull request by dontcallmedom**
closes #241

### CODE REVIEW

File: src/reporting/file-issue-for-review.js [hunk 1]
```diff
@@ -177,13 +193,17 @@ if (require.main === module) {
...
+	      await fs.unlink(issueFilename);
 	      execSync(`git checkout ${currentBranch}`);
 	    }
 	  }
 	}
       }
     }));
+    if (nolongerRelevantReports.size) {
+      console.log("The following reports are no longer relevant, deleting them", [...nolongerRelevantReports]);
+      [...nolongerRelevantReports].forEach(async issueFilename =>  await fs.unlink(issueFilename));
```
Comment (tidoust) [inline src/reporting/file-issue-for-review.js:205]: The `forEach` call won't wait for `unlink` calls to be done.

```suggestion
      for (const issueFilemane of nolongerRelevantReports) {
        await fs.unlink(issueFilename);
      }
```

**Review by tidoust — CHANGES_REQUESTED** [review]
The files are deleted from the local filesystem but isn't the goal to commit and push these changes onto the repo?

**Comment by dontcallmedom** [comment]
> The files are deleted from the local filesystem but isn't the goal to commit and push these changes onto the repo?

The `--update` mode of the CLI is at the moment not interacting with git on purpose - this is left at the discretion of the local user to review which changes should be committed / pushed.

The reasoning for that design is that `--update` is mostly meant for a manual processing approach (vs a CI based one), and so leaving more room to the user to review & decide if the results of the update process make sense feels more appropriate.

(and it is meant for a manual processing as I'm, probably naively, hoping we won't get back in a similar situation of reports that are pending submissions for several months)

**Review by tidoust — APPROVED** [review]
I see. The update mode does not update the repository. What a nice name for this mode ;)

**Pull request merged**